sklearn/svm/base.py

```diff
@@ -482,8 +482,8 @@ def _validate_for_predict(self, X):
     @property
     def coef_(self):
         if self.kernel != 'linear':
-            raise ValueError('coef_ is only available when using a '
-                             'linear kernel')
+            raise AttributeError('coef_ is only available when using a '
```

Comment:
Should we define a `class ParameterError(ValueError, AttributeError)`?

It would solve the issue of forward compatibility. I am a bit worried that too many exception classes will lead to these not being used. They need to be documented well.